### DISCUSSION TITLE
Is it possible to get subscription payload with select?

### DISCUSSION THREAD

**Discussion by Arenukvern** [supabase/supabase#7486] [Questions] [answered]
Hello everyone.

The problem (please note: this is extremely simplified example, business logic is completely different):

Imagine a table `User{id, name}`
And a table `UserLocation {id,  user_id,  city_name}`

If fetch UserLocation table using query select like below:
`"*, user:user_id (*) " `
It will result output:
`UserLocation {id,  user_id,  city_name, user: {id, name}}`

But in the same time, if to subscribe to table updates the payload will be always like this:
`UserLocation {id,  user_id,  city_name}`
And in this case, I will need to fetch user independently.

The question:
Is possible somehow to subscribe to the table's only rows and values, but receive data with applied select(`"*, user:user_id (*) " `) query?

Thanks for any help!

**Comment by GaryAustin1** [comment, marked as answer]
No.   You need to do the follow up select or imbed the needed data in the table you are monitoring with realtime.

**Reply by Arenukvern** [reply]
Thanks you.

It seems I missed also this recent discussion https://github.com/supabase/supabase/discussions/6059

**Reply by GaryAustin1** [reply]
Depending how much data you have changing in the "secondary" table, 6059 is quite painful.  If you only need a bit of profile info it my be better to use an update trigger on the user profile table to either update a column or two with your needed data in the main table you have realtime on.
Edit. This tigger idea also breaks down if you have lots of rows referencing the user now that I think about it. Easy enough to code, but could cause a lot of updates.  In my case I don’t care if user info changes, so a follow select works for me.

**Reply by Arenukvern** [reply]
Completely agree with you. The solution to update columns with a trigger works just fine.
